src/android/NativeRingtones.java

```diff
@@ -114,14 +130,24 @@ private boolean play(String ringtoneUri, boolean playOnce, int volume, int strea
             ringtoneSound.setAudioStreamType(streamType);
             if(volume >= 0) ringtoneSound.setVolume(volume * 0.01f, volume * 0.01f);
             ringtoneSound.prepare();
-
-            if (!playOnce) {
-                currentRingtone = ringtoneSound;
-            } else {
+            currentRingtone = ringtoneSound;
+            if (playOnce) {
                 ringtoneSound.setOnCompletionListener(new MediaPlayer.OnCompletionListener() {
                     public void onCompletion(MediaPlayer mp) {
                         mp.stop();
                         mp.release();
+                        if (completeCallbacks != null) {
```

Comment:
Copidao de aqui: https://github.com/we-are-Joinup/cordova-plugin-nativeaudio/blob/master/src/android/NativeAudio.java#L119